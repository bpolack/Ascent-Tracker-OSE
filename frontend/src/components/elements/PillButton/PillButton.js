import React, { Component } from 'react'

export class PillButton extends Component {
	render() {

		let style = 'text-gray-800 bg-white hover:text-white hover:bg-pink-500';
		if (this.props.varient === 'color') {
			style = 'text-white bg-indigo-500 hover:bg-pink-500';
		}
		else if (this.props.varient === 'dark') {
			style = 'bg-gray-800 text-white hover:text-white hover:bg-pink-500';
		}

		return (
			<a href={this.props.link} onClick={this.props.buttonClickEvent} className={`cursor-pointer inline-block leading-none mx-2 px-5 py-3 font-semibold transition-colors rounded-full ${style} ${this.props.className}`}>
				{this.props.children}
			</a>
		)
	}
}

export default PillButton
