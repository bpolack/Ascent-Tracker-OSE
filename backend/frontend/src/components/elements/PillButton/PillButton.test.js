import React from 'react';
import { shallow } from 'enzyme';
import PillButton from './PillButton';

describe('PillButton', () => {
	
	it("Renders correctly without crashing", () => {
		const wrapper = shallow(<PillButton link="https://polackdevelopment.com" varient="color">Test Button</PillButton>);
		expect(wrapper).toMatchSnapshot();
	});

	describe('When user clicks PillButton', () => {
		it('calls the correct onClick function', () => {
			const onButtonClickMock = jest.fn();
			
			const wrapper = shallow(
				<PillButton buttonClickEvent={onButtonClickMock} varient="color">Test Button</PillButton>,
			);

			const buttonElement = wrapper.find('a'); 
			buttonElement.simulate('click'); 

			expect(onButtonClickMock).toHaveBeenCalledTimes(1);
		});
	});
});