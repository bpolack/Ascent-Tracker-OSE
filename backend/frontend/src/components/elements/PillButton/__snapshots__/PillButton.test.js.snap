// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PillButton Renders correctly without crashing 1`] = `ShallowWrapper {}`;
